doc/dev/developer_guide/debugging-gdb.rst

```diff
@@ -0,0 +1,47 @@
+GDB - The GNU Project Debugger
+==============================
+
+`The GNU Project Debugger (GDB) <https://www.sourceware.org/gdb>`_ is
+a powerful tool which allows you to analyze the execution flow.
+GDB can help to find bugs, uncover crash errors or track the
+source code during execution.
```

Comment:
I would add a short sentence or two to this paragraph explaining how GDB can be applied to the Ceph project. You already explain below that it can be used to debug Teuthology tests, but I would also include that info briefly at the top here, as well as an indication that it is a good tool to use in development clusters.